docs/authentication/enterprise-connections/oidc/custom-provider.mdx

```diff
@@ -0,0 +1,86 @@
+
+<Steps>
+  ### Set up an enterprise connection in Clerk
+
+  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
+  1. Select **Add connection** and select **For specific domains**.
+  1. Under **Third party**, select **OIDC**.
+  1. Add the **Name** of the connection.
+  1. Add the **Key** of the provider. This is the provider's unique identifier (cannot be changed after creation).
+  1. Add the **Specific Domain** that you want to allow this connection for. This is the domain of the users you want to allow to sign in to your app.
+  1. Select **Add connection**. You will be redirected to the connection's configuration page.
+
+  The next steps involve configuring both your IdP and Clerk. Keep both dashboards open as you'll need to reference them.
+
+  ### Configure your IdP
+
+  1. If necessary, create a new app in your IdP.
+  1. In the connection's configuration page in the Clerk Dashboard, copy the **Authorized redirect URI**.
+  1. Add the value to your IdP's whitelisted URLs.
+
+  ### Set the Discovery Endpoint, Client ID, and Client Secret in Clerk
+
+  1. In your IdP settings, copy your application's **Discovery Endpoint**, **Client ID**, and **Client Secret**.
+  1. In the connection's configuration page in the Clerk Dashboard, paste these values in their respective fields.
+  1. Add the minimum required scopes based on the IdP's documentation. Common OIDC scopes include `openid`, `email`, and `profile`.
+  1. Select **Save**.
+
+  ### Configure attribute mapping (optional)
+
+  If your provider returns claims in a non-standard format, use the **Attribute Mapping** section on the connection's configuration page to adjust the mapping of Clerk's user properties to match the IdP's claim attributes. Important Clerk user properties include `email`, `firstName`, and `lastName`.
+
+  > [!CAUTION]
+  > OIDC Enterprise connections require the `email_verified` claim to verify email ownership. If the IdP doesn't return this claim, you can set its default value of `true`. However, ensure you fully trust the IdP to prevent exploits like [nOAuth](https://www.descope.com/blog/post/noauth).
+
+  ### Allow additional identifiers (optional)
+
+  User profile information is sourced from the IdP. To allow users to add new identifiers (e.g., email address or phone number) to their profiles:
+
+  1. Navigate to the **Advanced** tab of the connection's configuration page.
+  1. Enable **Allow additional identifiers**.
+  1. Select **Save**.
+
+  ### Enable the connection for Clerk
+
+  To make the connection available for your users to authenticate with:
+
+  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
+  1. Select the OIDC connection.
+  1. Toggle on **Enable connection**.
+  1. Select **Save**.
```

Comment:
they should already have the dashboard open, with the sso connections page.
we use this copy in other guides:

```suggestion
  To make the connection available for your users to authenticate with:

  1. Navigate back to the Clerk Dashboard where you should still have the connection's configuration page open. If not, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page and select the connection.
  1. At the top of the page, toggle on **Enable connection** and select **Save**.
```